logging/logging-processor/src/main/java/org/infinispan/logging/processor/InfinispanLoggingProcessor.java

```diff
@@ -0,0 +1,116 @@
+import java.util.stream.Stream;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic;
+import javax.tools.StandardLocation;
+import javax.xml.stream.XMLStreamException;
+
+import org.infinispan.logging.processor.report.XmlReportWriter;
+import org.jboss.logging.processor.apt.ProcessingException;
+import org.kohsuke.MetaInfServices;
+/**
+ * InfinispanLoggingProcessor.
+ *
+ * @author Durgesh Anaokar
+ * @since 13.0
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@SupportedAnnotationTypes({InfinispanLoggingProcessor.DESCRIPTION})
+@MetaInfServices(Processor.class)
+public class InfinispanLoggingProcessor extends AbstractProcessor {
+   static final String DESCRIPTION = "org.infinispan.logging.annotations.Description";
+
+   @Override
+   public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+      for (TypeElement annotation : annotations) {
+            createReportXml(roundEnv, annotation);
+      }
+      return true;
+   }
+
+   private void createReportXml(RoundEnvironment roundEnv, TypeElement annotation){
+         Set<? extends Element> annotatedElements = roundEnv.getElementsAnnotatedWith(annotation);
+         Map<String, List<Element>> mapWithQualifiedName = getMapWithQualifiedName(annotatedElements);
+         for (Map.Entry<String, List<Element>> entry : mapWithQualifiedName.entrySet()) {
+            Element e = entry.getValue().get(0);
+            String qualifiedName = e.getEnclosingElement().toString();
+            Element enclosing = e.getEnclosingElement();
+            String packageName = processingEnv.getElementUtils().getPackageOf(enclosing).getQualifiedName().toString();
+            String simpleName = enclosing.getSimpleName().toString();
+            try (BufferedWriter bufferedWriter = createWriter(packageName, simpleName + ".xml");
+               XmlReportWriter reportWriter = new XmlReportWriter(bufferedWriter)) {
+               reportWriter.writeHeader(qualifiedName);
+               for (Element element : entry.getValue()) {
+                  reportWriter.writeDetail(element);
+               }
+               reportWriter.writeFooter();
+            } catch (ProcessingException ex) {
+               final AnnotationMirror a = ex.getAnnotation();
+               final AnnotationValue value = ex.getAnnotationValue();
+               error(e, "Error while processing the report: for annotation" + a.toString() + " for value: "
+                     + value.toString(), e);
```

Comment:
We shouldn't catch `ProcessingException`, instead we should let the exception propogate as it's unchecked and compilation will fail.